include/sys/zfs_ioctl.h

```diff
@@ -160,16 +160,35 @@ typedef enum dmu_send_resume_token_version {
  * cannot necessarily be received as a clone correctly.
  */
 #define	DRR_FLAG_FREERECORDS	(1<<2)
+/*
+ * When DRR_FLAG_SPILL_BLOCK is set it indicates the DRR_SPILL_BLOCK flag
+ * is valid in the send stream.  A single bit was shared for this purpose
+ * because drr_begin->drr_flags is only a uint8_t.
```

Comment:
> drr_begin->drr_flags is uint32_t (drr_object->drr_flags is uint8_t)

I can try and make this comment clearer.  There is plenty of room in `drr_begin->drr_flags`.  My concern was with `drr_object->drr_flags` and `drr_spill->flags` which are both a uint8_t.  See next comment.

> I don't see where this bit-sharing is taking place?

The sharing referred to is that `DRR_SPILL_BLOCK` can be set for both `drr_object` and `drr_spill` records.  But it has a different meaning for each record type.  Are we OK with this, or should we use another bit and define a `DRR_HAS_SPILL_BLOCK` for drr_object flags and a `DRR_EXTRA_SPILL_BLOCK` for drr_spill flags.

> Is the past tense ("was") relevant here?

No, I'll update it.

Reply:
Personally I'm ok with the current scheme(using the same bit in each type, with different meanings), but the language in this comment does need some clarification. I'll provide my rough draft below, you can pick and choose bits you like, or ignore it if you prefer:

```
When DRR_FLAG_SPILL_BLOCK is set it indicates the DRR_SPILL_BLOCK flag is meaningful in the send stream.

When DRR_FLAG_SPILL_BLOCK is set, DRR_OBJECT records will have DRR_SPILL_BLOCK set if and only if they should have a spill block (either an existing one, or a new one in the send stream).  When clear the object does not have a spill block and any existing spill block should be freed.

Similarly, when DRR_FLAG_SPILL_BLOCK is set, DRR_SPILL records will have DRR_SPILL_BLOCK set if and only if they were included for backward compatibility purposes, and can be safely ignored by new version of zfs receive.  Previous versions of ZFS which do not understand the DRR_FLAG_SPILL_BLOCK will process this record and recreate any missing spill blocks.

DRR_SPILL_BLOCK has different meanings in different record types because the meanings are closely linked, there is per-record-type specific behavior, and there is only a small amount of space (8 bits) in drr_flags in the drr_object and drr_spill structures.
```

Reply:
Thanks, I opted to use your updated comment.